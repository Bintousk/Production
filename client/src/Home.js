import React, {useEffect, useState} from 'react'
import "./Home.css"

// For fetching data
import axios from 'axios'

import Card from './Card';
import Trash from './Trash';

const Home = () => {

    const [items, setItems] = useState()

    useEffect(()=>{
        axios.get('http://localhost:5002/items/read')
        .then(res => {
            setItems(res.data)
        })
    },[])



    return (
        <div>
            <h1 className='title'>Items List</h1>
            <div className="container">
                {items && items.map(item => ! item.deletionComment ? (
                    <Card type="row" key={item._id} 
                    item={item} allItems={items} 
                    setItems={setItems}/>
                ):
                <React.Fragment key={item._id}></React.Fragment>
               
                )}
                
                <Card type="add" setItems={setItems}/>
            </div>
            <h1 className='title'>Garbage List</h1>
            <div className="container">
                {items && items.map(item =>  item.deletionComment ? (
                    <Trash type="row" key={item._id} 
                    item={item} allItems={items} 
                    setItems={setItems}/>
                ):
                <React.Fragment key={item._id}></React.Fragment>
               
                )}
                
               
            </div>
        </div>
    )
}

export default Home
