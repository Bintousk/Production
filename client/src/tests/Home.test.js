import React from "react";

import { create } from "react-test-renderer";

import Home from "../Home";




describe("Test for Home.js", () => {
   
   test('Matches the snapshot', () => { 
   const component = create(<Home/>);
    expect(component.toJSON()).toMatchSnapshot();
   })

})
