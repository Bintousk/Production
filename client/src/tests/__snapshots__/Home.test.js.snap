// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for Home.js Matches the snapshot 1`] = `
<div>
  <h1
    className="title"
  >
    Items List
  </h1>
  <div
    className="container"
  >
    <div>
      <div
        className="addcards"
        onClick={[Function]}
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M13 7h-2v4H7v2h4v4h2v-4h4v-2h-4V7zm-1-5C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8z"
          />
        </svg>
      </div>
    </div>
  </div>
  <h1
    className="title"
  >
    Garbage List
  </h1>
  <div
    className="container"
  />
</div>
`;

exports[`should show the cards section 1`] = `
<div>
  <h1
    className="title"
  >
    Items List
  </h1>
  <div
    className="container"
  >
    <div>
      <div
        className="addcards"
        onClick={[Function]}
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M13 7h-2v4H7v2h4v4h2v-4h4v-2h-4V7zm-1-5C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8z"
          />
        </svg>
      </div>
    </div>
  </div>
  <h1
    className="title"
  >
    Garbage List
  </h1>
  <div
    className="container"
  />
</div>
`;
